### PR TITLE
Useful enhancements for running the pipeline

### PR DESCRIPTION
**Pull request by jrs65**
- fix(config): format string bugs
- feat(template-run): add option to submit templat jobs to the queue
- feat(runner,mpiutil): allow control of MPI abort on exception

### CODE REVIEW

File: caput/scripts/runner.py [hunk 1]
```diff
@@ -153,7 +153,11 @@ def run(configfile, loglevel, profile, profiler):
     ),
     metavar="<NAME>=<VALUE>[,<VALUE2>...]",
 )
-def template_run(templatefile, var):
+@click.option(
+    "--submit/--nosubmit", default=False, help="Submit into the batch queue"
+)
+@click.pass_context
+def template_run(ctx, templatefile, submit, var):
     """Run a pipeline immediately from the given TEMPLATEFILE.
```
Comment (anjakefala) [inline caput/scripts/runner.py:172]: > From the given TEMPLATEFILE, run a pipeline immediately (default), or submit it into the batch queue (`--nosubmit`).

Comment (anjakefala) [reply caput/scripts/runner.py:172]: Docstring rewording suggestion ^

File: caput/scripts/runner.py [hunk 2]
```diff
@@ -190,8 +194,11 @@ def template_run(templatefile, var):
             tfh.write(expanded_string)
             tfh.flush()
 
-            # Run the pipeline
-            Manager.from_yaml_file(tfh.name).run()
+            # Run or queue the pipeline
+            if queue:
+                ctx.invoke(queue, configfile=tfh.name)
```
Comment (anjakefala) [inline caput/scripts/runner.py:210]: Do you need to pass `submit=True`?

File: caput/scripts/runner.py [hunk 3]
```diff
@@ -190,8 +194,11 @@ def template_run(templatefile, var):
             tfh.write(expanded_string)
             tfh.flush()
 
-            # Run the pipeline
-            Manager.from_yaml_file(tfh.name).run()
+            # Run or queue the pipeline
+            if queue:
```
Comment (anjakefala) [inline caput/scripts/runner.py:209]: Where does `queue` come from?

**Comment by anjakefala** [comment]
I have some questions about 5f7f8c8 that might be the result of my relatively poor understanding of `click` and contexts.

Everything else looks great! I also do not think it will conflict too badly with my PR. Thanks for fixing the format strings

File: caput/scripts/runner.py (hunk 3 above)
Comment (jrs65) [reply caput/scripts/runner.py:209]: Haha. Good spot. I already fixed that one but I haven't pushed the commit. I found it when it submitted a bunch of jobs into the queue that I had wanted it to run immediately!

File: caput/scripts/runner.py (hunk 2 above)
Comment (jrs65) [reply caput/scripts/runner.py:210]: Yeah, the issue is that it should be `if submit` !

File: caput/scripts/runner.py (hunk 3 above)
Comment (anjakefala) [reply caput/scripts/runner.py:209]: Ah! Okay, once that is pushed, we can merge this in. =)

**Review by anjakefala — APPROVED** [review]

**Pull request merged**